README.md

```diff
@@ -54,14 +72,176 @@ Solana cli.
 
 **Example Use Case:** A script that needs to authenticate users offchain, without relying on a browser extension.
 
+## Authentication Flow
+
+The authentication flow is very similar for all methods. The flow is as follows:
+
+1. The user:
+    1. initiates the authentication flow
+    2. providers their Solana public key
+    3. sends this to the application API
+2. The application
+    1. Creates a verification message using the public key
+    2. Sends the verification message to user
+3. The user:
+    1. signs the verification message
+    2. sends the signed message to the application API
+4. The application API:
+    1. Verifies the signed message
+    2. Returns the authentication result
+5. The user:
+    1. Has now verified that they can sign messages with the provided public key.
+
```

Comment:
The capitalization of the first word of the bullet points is inconsistent. (I prefer upper case)